### PR TITLE
Adding force https metadata key

### PR DESCRIPTION
**Pull request by TenSt**
# Description

Added `ForceHTTPS` key to metadata and simple if statement that will handle forcing https in the Location header of the final redirect from oauth2 middleware. 

## Issue reference

Reference to the issue this PR will close: #486

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests - there were no tests for this middleware, so I didn't created any new ones.
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#836
* [x] Tested in the k8s environment with the latest stable dapr release v0.10.0.

### CODE REVIEW

File: middleware/http/oauth2/oauth2_middleware.go [hunk 1]
```diff
@@ -81,6 +82,9 @@ func (m *Middleware) GetHandler(metadata middleware.Metadata) (func(h fasthttp.R
 			} else {
 				authState := session.GetString(savedState)
 				redirectURL := session.GetString(redirectPath)
+				if meta.ForceHTTPS == "true" {
```
Comment (youngbupark) [inline middleware/http/oauth2/oauth2_middleware.go:85]: I would change ForceHTTP to lowercase first and compare with `"true"`

File: middleware/http/oauth2/oauth2_middleware.go [hunk 2]
```diff
@@ -81,6 +82,9 @@ func (m *Middleware) GetHandler(metadata middleware.Metadata) (func(h fasthttp.R
...
+					redirectURL = "https://" + string(ctx.Request.Host()) + redirectURL
```
Comment (youngbupark) [inline middleware/http/oauth2/oauth2_middleware.go:86]: Can we make "http://" const?

File: middleware/http/oauth2/oauth2_middleware.go [hunk 3]
```diff
@@ -81,6 +82,9 @@ func (m *Middleware) GetHandler(metadata middleware.Metadata) (func(h fasthttp.R
...
+				}
```
Comment (youngbupark) [inline middleware/http/oauth2/oauth2_middleware.go:87]: Please add unit-test for this change.

**Review by youngbupark — CHANGES_REQUESTED** [review]
Thanks for your contribution! I left some comments. please add unit-test as well.

File: middleware/http/oauth2/oauth2_middleware.go (hunk 1 above)
Comment (TenSt) [reply middleware/http/oauth2/oauth2_middleware.go]: Done.

File: middleware/http/oauth2/oauth2_middleware.go (hunk 2 above)
Comment (TenSt) [reply middleware/http/oauth2/oauth2_middleware.go]: Done. Added const `https`.

File: middleware/http/oauth2/oauth2_middleware.go (hunk 3 above)
Comment (TenSt) [reply middleware/http/oauth2/oauth2_middleware.go:88]: Working on it.

Comment (TenSt) [reply middleware/http/oauth2/oauth2_middleware.go:88]: Hi @youngbupark. Could you please give me advice on how unit test for this should be done? This is a Authorization Code Flow of OAuth2 protocol which should be done via browser to pass user credentials directly to Authorization Provider without getting them to the app. When I was testing it locally I've setup whole infrastructure in order to do that. There is no existing unit tests for this middleware, so I can't mimic existing tests or extend them. Maybe I'm looking at the wrong direction and don't see some obvious ways to do this. Advice or help would be appreciated.

Comment (youngbupark) [reply middleware/http/oauth2/oauth2_middleware.go:88]: Thanks for your investigation and your time. The initial ask is to test the below code by separating it as separate method or using mock:

```
redirectURL := session.GetString(redirectPath)
				if meta.ForceHTTPS == "true" {
					redirectURL = "https://" + string(ctx.Request.Host()) + redirectURL
				}
```

If it is difficult to write the test without the actual backend service support and you validate it locally, I am ok to merge it now.  @yaron2 any objection?

JFYI - We are planning to have conformance test. we will provide the common test infra for middleware test later. @mukundansundar

**Review by youngbupark — APPROVED** [review]

**Comment by yaron2** [comment]
Thanks for this @TenSt

**Comment by TenSt** [comment]
@yaron2 it was my pleasure

**Pull request merged**